Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift

```diff
@@ -156,22 +183,51 @@ class TimeFrameTimeZoneChallengeViewController: UIViewController {
             } else {
                 if doneValue != "" {
                     tempArr = self.generateTimeZoneArray(isFrom: self.isFromButton, fromToValue: self.zonesArray[self.zonesArray.endIndex - 1], withDoneValue: doneValue)
-                    self.zonesArray[self.zonesArray.endIndex - 1] = tempArr!
+                    
+                    if !self.isFromButton {
+                        let arr = tempArr!.dashRemoval()
```

Comment:
In this caseforce unwrapped it because it will never be nil. Will take care of force unwrap when I do code refactor.